code/_onclick/ai.dm

```diff
@@ -33,9 +33,9 @@
 
 	if(control_disabled || stat)
 		return
-		
+
 	var/turf/pixel_turf = get_turf_pixel(A)
```

Comment:
replaced 37-42 with:

```
	var/turf/pixel_turf = get_turf_pixel(A)
	var/turf_visible
	if(pixel_turf)
		turf_visible = cameranet.checkTurfVis(pixel_turf)
		if(!turf_visible)
			if(istype(loc, /obj/item/device/aicard) && (pixel_turf in view(client.view, loc)))
				turf_visible = TRUE
			else
				if(pixel_turf.obscured)
					log_admin("[key_name_admin(src)] might be running a modified client! (failed checkTurfVis on AI click of [A]([COORD(pixel_turf)])")
					message_admins("[key_name_admin(src)] might be running a modified client! (failed checkTurfVis on AI click of [A]([ADMIN_COORDJMP(pixel_turf)]))")
					send2irc_adminless_only("NOCHEAT", "[key_name(src)] might be running a modified client! (failed checkTurfVis on AI click of [A]([COORD(pixel_turf)]))")
				return
```

TG made a number of tweaks, fixes, and updates to this so it's more accurate.